### PR TITLE
RealmRoles/Global roles field to header

### PR DESCRIPTION
**Pull request by BlueWaves2**
## Description of changes

Adding
- new field in header for logged-in user for realmRoles (keycloak)
- renamed Topic to DaprTopic (for Rick) 
- fixed Tests to contain changes
- refactored Testcode
- added Utility Tetsclass to generated testobjects for Testcases

  
## How has this been tested?
Please describe the test strategy you followed.

- [X] automated unit test
- [ ] automated integration test
- [ ] automated acceptance test
- [ ] manual, exploratory test
    
## Checklist before requesting a review
- [X] My code follows the [coding guidelines](https://github.com/IT-REX-Platform/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] My code fulfills all acceptance criteria
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] The test coverage (line and branch) is reasonably high, especially on Service classes and other classes with complex logic
- [X] I have made corresponding changes to the documentation
- [X] I have added explanation of architectural decision and rationales to [wiki/adr](https://github.com/IT-REX-Platform/wiki/tree/main/adr)
- [X] I have updated the changes in the ticket description

## Checklist for reviewer
- [ ] The code works and does not throw errors
- [ ] The code is easy to understand and there are no confusing parts
- [ ] The code follows the [coding guidelines](https://github.com/IT-REX-Platform/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [ ] The code change accomplishes what it is supposed to do
- [ ] I cannot think of any use case in which the code does not behave as intended
- [ ] The added and existing tests reasonably cover the code change
- [ ] I cannot think of any test cases, input or edge cases that should be tested in addition
- [ ] Description of the change is included in the documentation

### CODE REVIEW

File: src/test/java/de/unistuttgart/iste/gits/common/TestUserUtil.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+import de.unistuttgart.iste.gits.common.user_handling.LoggedInUser;
+
+import java.time.OffsetDateTime;
+import java.util.List;
+import java.util.UUID;
+
+public class TestUserUtil {
+
+    public static LoggedInUser generateUser(final LoggedInUser.CourseMembership... courseMemberships){
```
Comment (PaulBredl) [inline src/test/java/de/unistuttgart/iste/gits/common/TestUserUtil.java:11]: createUserWithMemberships

(optional todo)

**Review by PaulBredl — APPROVED** [review]
optional todo
